### PR TITLE
Feature/room versioning

### PR DESCRIPTION
**Pull request by ganfra**
Implement Room versioning : https://github.com/vector-im/riot-android/issues/2441

### CODE REVIEW

File: matrix-sdk/src/main/java/org/matrix/androidsdk/rest/model/RoomTombstoneContent.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2016 OpenMarket Ltd
```
Comment (manuroe) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/rest/model/RoomTombstoneContent.java]: Copyright 2018 is enough.

File: matrix-sdk/src/main/java/org/matrix/androidsdk/rest/model/RoomCreateContent.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 OpenMarket Ltd
```
Comment (manuroe) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/rest/model/RoomCreateContent.java]: Copyright 2018 is enough.

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java [hunk 3]
```diff
@@ -544,17 +545,16 @@ public String getVisibility() {
...
-        RoomState state = getState();
-        String membership = null;
-
-        RoomMember selfMember = state.getMember(mMyUserId);
+        return hasMembership(RoomMember.MEMBERSHIP_INVITE);
+    }
 
-        if (null != selfMember) {
-            membership = selfMember.membership;
+    public boolean hasMembership(@NonNull final String membership) {
```
Comment (bmarty) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java]: Please document this public method

File: matrix-sdk/src/main/java/org/matrix/androidsdk/data/RoomState.java [hunk 4]
```diff
@@ -822,6 +827,34 @@ public boolean isEncrypted() {
         return null != algorithm;
     }
 
+    /**
+     * @return true if the room is versioned
```
Comment (bmarty) [inline matrix-sdk/src/main/java/org/matrix/androidsdk/data/RoomState.java]: Maybe we should explain what it means if this method return true

**Pull request merged**